common/lib/kernel/devices/ip_device.cpp

```diff
@@ -0,0 +1,53 @@
+/* IpDevice
+ *
+ * Author: Pascal Bauer <pascal.bauer@rwth-aachen.de>
+ *
+ * SPDX-FileCopyrightText: 2023-2024 Pascal Bauer <pascal.bauer@rwth-aachen.de>
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <filesystem>
+#include <regex>
+#include <stdexcept>
+
+#include <villas/kernel/devices/ip_device.hpp>
+
+using villas::kernel::devices::IpDevice;
+
+IpDevice IpDevice::from(const std::filesystem::path unsafe_path) {
+  if (!is_path_valid(unsafe_path))
+    throw std::runtime_error(
```

Comment:
I think we already have our own Exception classes which use `fmt::format` to format the exception message.